pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1202,6 +1211,22 @@ type Sigv4 struct {
 	RoleArn string `json:"roleArn,omitempty"`
 }
 
+// AzureAD defines the configuration for remote write's azuread parameters.
+// +k8s:openapi-gen=true
+type AzureAD struct {
+	// The Azure Cloud. Options are 'AzurePublic', 'AzureChina', or 'AzureGovernment'. If blank, AzurePublic is used.
+	Cloud string `json:"cloud,omitempty"`
+	// ManagedIdentity defines the Azure User-assigned Managed identity.
+	ManagedIdentity *ManagedIdentity `json:"managedIdentity"`
+}
+
+// ManagedIdentity defines the Azure User-assigned Managed identity.
+// +k8s:openapi-gen=true
+type ManagedIdentity struct {
+	// The secret or configmap containing the client id
+	ClientID SecretOrConfigMap `json:"clientId"`
```

Comment:
From the prometheus configuration doc(https://prometheus.io/docs/prometheus/latest/configuration/configuration/), the client_id was defined as <string> instead of <secret>, so I think is depends on what user think about this.

Reply:
If it's really not sensitive, I'd go with a string but I'm not sure yet.

Reply:
Hi @heylongdacoder , I've searched a bit (googled it :P) and noticed that AzureAD's Client ID is not sensitive at all, it is public data. 

Could we simplify here and just use an string instead of SecretOrConfigMap?

Reply:
thanks for the confirmation, I have changed it to string 😄

Reply:
ClientID is usually not considered sensitive information, sorry for my late reply.